libs/components/filter-bar/testing/src/modules/filter-bar/filter-bar-harness.ts

```diff
@@ -0,0 +1,106 @@
+import { HarnessPredicate } from '@angular/cdk/testing';
+import { SkyComponentHarness } from '@skyux/core/testing';
+import { SkySelectionModalHarness } from '@skyux/lookup/testing';
+
+import { SkyFilterBarHarnessFilters } from './filter-bar-harness-filters';
+import { SkyFilterBarItemHarness } from './filter-bar-item-harness';
+import { SkyFilterBarItemHarnessFilters } from './filter-bar-item-harness-filters';
+
+/**
+ * Harness for interacting with a filter bar component in tests.
+ */
+export class SkyFilterBarHarness extends SkyComponentHarness {
+  /**
+   * @internal
+   */
+  public static hostSelector = 'sky-filter-bar';
+
+  #documentRootLocator = this.documentRootLocatorFactory();
+
+  #getClearFiltersButton = this.locatorForOptional(
+    '.sky-filter-bar-clear-filters',
+  );
+  #getFilterPickerButton = this.locatorForOptional(
+    '.sky-filter-bar-filter-picker',
+  );
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a
+   * `SkyFilterBarHarness` that meets certain criteria.
+   */
+  public static with(
+    filters: SkyFilterBarHarnessFilters,
+  ): HarnessPredicate<SkyFilterBarHarness> {
+    return SkyFilterBarHarness.getDataSkyIdPredicate(filters);
+  }
+
+  /**
+   * Clicks the clear filters button.
+   */
+  public async clickClearFilters(): Promise<void> {
+    const button = await this.#getClearFiltersButton();
+    if (!button) {
+      throw new Error('Clear filters button not found');
+    }
+    await button.click();
+  }
+
+  /**
+   * Gets a harness for a specific filter bar item that meets certain criteria.
+   */
+  public async getItem(
+    filter: SkyFilterBarItemHarnessFilters,
+  ): Promise<SkyFilterBarItemHarness> {
+    return await this.locatorFor(SkyFilterBarItemHarness.with(filter))();
+  }
+
+  /**
+   * Gets an array of all filter bar items.
+   */
+  public async getItems(
+    filters?: SkyFilterBarItemHarnessFilters,
+  ): Promise<SkyFilterBarItemHarness[]> {
+    const items = await this.locatorForAll(
+      SkyFilterBarItemHarness.with(filters || {}),
+    )();
+
+    if (filters && items.length === 0) {
+      throw new Error(
+        `Unable to find any filter bar items with filter(s): ${JSON.stringify(filters)}`,
+      );
+    }
+
+    return items;
+  }
+
+  /**
+   * Checks if the filter bar has active filters.
+   */
+  public async hasActiveFilters(): Promise<boolean> {
+    const clearButton = await this.#getClearFiltersButton();
+    return !!clearButton;
+  }
+
+  /**
+   * Checks if the filter picker button is visible.
+   */
+  public async hasFilterPicker(): Promise<boolean> {
+    const pickerButton = await this.#getFilterPickerButton();
+    return !!pickerButton;
+  }
+
+  /**
+   * Clicks the filter picker button.
```

Comment:
can we add more clarity as to what we are sending back to them, bc this function does more than just click the filter picker button.